packages/terra-functional-testing/src/services/wdio-terra-service.js

```diff
@@ -147,30 +147,52 @@ class TerraService {
-
-        const commentsResult = await octokit.request(`GET /repos/${repoName[0]}/${repoName[1]}/issues/${this.serviceOptions.issueNumber}/comments`);
+        const message = [
+          ':warning: :bangbang: **WDIO MISMATCH** \n\n',
+          `Check that screenshot change is intended at: ${this.serviceOptions.buildUrl} \n\n`,
+          'If screenshot change is intended, remote reference screenshots will be updated upon PR merge. \n',
+          'If screenshot change is unintended, please fix screenshot issues before PR merge to prevent them from being uploaded. \n\n',
+          'Note: This comment only appears the first time a screenshot mismatch is detected on a PR build, ',
+          'future builds will need to be checked for unintended screenshot mismatchs.',
+        ].join('');
+
+        const commentsUrl = `/repos/${repoName[0]}/${repoName[1]}/issues/${this.serviceOptions.issueNumber}/comments`;
+        const commentsResult = await octokit.request(`GET ${commentsUrl}`);
         const existingComment = commentsResult.data.find((comment) => comment.body === message);
 
         if (!existingComment) {
-          const postCommentResult = await octokit.request(`POST /repos/${repoName[0]}/${repoName[1]}/issues/${this.serviceOptions.issueNumber}/comments`, {
+          const postCommentResult = await octokit.request(`POST ${commentsUrl}`, {
             body: message,
           });
           if (postCommentResult.status !== 200) {
             throw Error(`Error posting issue comment. Status code: ${postCommentResult.status}`);
           }
         }
-      } else if (this.serviceOptions.useRemoteReferenceScreenshots && !this.serviceOptions.buildBranch.match(BUILD_BRANCH.pullRequest) && this.serviceOptions.buildType === BUILD_TYPE.branchEventCause) {
-        const screenshotConfig = config.getRemoteScreenshotConfiguration ? config.getRemoteScreenshotConfiguration(config.screenshotsSites, this.serviceOptions.buildBranch) : {};
+      } else if (
+        // Branch event here means the PR has been merged. We will upload the screenshots to the base branch of the PR.
+        // The PR's base branch is the merge target of the PR, so if the PR was from my-feature to master, base branch's
+        // ref is master.
+        this.serviceOptions.useRemoteReferenceScreenshots
+        && !this.serviceOptions.buildBranch.match(BUILD_BRANCH.pullRequest)
+        && this.serviceOptions.buildType === BUILD_TYPE.branchEventCause
+        && config.getRemoteScreenshotConfiguration
+      ) {
+        const pr = await octokit.request(`GET /repos/${repoName[0]}/${repoName[1]}/pulls/${this.serviceOptions.issueNumber}/`);
+        const prBaseBranch = pr.data.base.ref;
+        const screenshotConfig = config.getRemoteScreenshotConfiguration(config.screenshotsSites, prBaseBranch);
```

Comment:
Part of the acceptance criteria for this is we would also need to do something similar to this in `onPrepare` so that we know which targeted branch to use to _download_ the screenshots. After the PR is issued and the user decides to change the current base branch to a different base branch, this service would then use the updated base branch for downloading the screenshots. I would think `pr.data.base.ref` would reflect this if the base target branch does get changed.

Reply:
oh shoot! Ok I'll add that, thanks.

Reply:
Added as of eb70843f945d3e964464c5f483c67b051faa6a19.